pennylane/transforms/draw.py

```diff
@@ -152,6 +152,148 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def draw(
+    qnode,
+    wire_order=None,
+    show_all_wires=False,
+    decimals=None,
+    max_length=100,
+    expansion_strategy=None,
+):
+    """Create a function that draws the given qnode.
+
+    Args:
+        qnode (.QNode): the input QNode that is to be drawn.
+        wire_order (Sequence[Any]): the order (from top to bottom) to print the wires of the circuit
+        show_all_wires (bool): If True, all wires, including empty wires, are printed.
+        decimals (int): How many decimal points to include when formatting operation parameters.
+            Default ``None`` will omit parameters from operation labels.
+        max_length (int): Maximum string width (columns) when printing the circuit
+        expansion_strategy (str): The strategy to use when circuit expansions or decompositions
+            are required.
+
+            - ``gradient``: The QNode will attempt to decompose
+              the internal circuit such that all circuit operations are supported by the gradient
+              method.
+
+            - ``device``: The QNode will attempt to decompose the internal circuit
+              such that all circuit operations are natively supported by the device.
+
+
+    Returns:
+        A function that has the same argument signature as ``qnode``. When called,
+        the function will draw the QNode.
+
+    **Example**
+
+    .. code-block:: python3
+
+        @qml.qnode(qml.device('lightning.qubit', wires=2))
+        def circuit(a, w):
+            qml.Hadamard(0)
+            qml.CRX(a, wires=[0, 1])
+            qml.Rot(*w, wires=[1])
+            qml.CRX(-a, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+    >>> print(qml.draw(circuit)(a=2.3, w=[1.2, 3.2, 0.7]))
+    0: ──H─╭C───────╭C──┤ ╭<Z@Z>
+    1: ────╰RX──Rot─╰RX─┤ ╰<Z@Z>
+
+    .. UsageDetails::
+
+
+    By default, parameters are omitted. By specifying the ``decimals`` keyword, parameters
+    are displayed to the specified precision. Matrix-valued parameters are never displayed.

```

Comment:
Here also it might be helpful not necessarily printing the matrix but at least label it M1, M2, etc which helps distinguish the different gates and identify which ones use the same matrix parameter.

Reply:
I have a plan to add support for this in the future. If necessary, I can wait on this PR till  I figure that out.

Reply:
I'm ok with separate PRs, that way things keep moving forward :)